lib/linguist/blob_helper.rb

```diff
@@ -337,10 +337,11 @@ def tm_scope
 
     # Internal: Should this blob be included in repository language statistics?
     def include_in_language_stats?
-      !vendored? &&
-      !documentation? &&
-      !generated? &&
-      language && DETECTABLE_TYPES.include?(language.type)
+      defined?(detectable?) && !detectable?.nil? ? language && detectable? :
```

Comment:
This is going to need refining as it's too greedy and overrides any of the subsequent attributes in this method.

For example, creating a repo containing:

```
index.html
docs/index.html
vendor/index.html
vendor/docs/index.html
```

... and analysing using the default attributes results in:

```
$ ../seppestas-linguist/bin/linguist --breakdown .
100.00% HTML

HTML:
index.html

$
```

If we then implement a `.gitattributes` like this:

```
vendor/* linguist-vendored=false
vendor/docs/* linguist-documentation
```

... and run again, we get:

```
$ ../seppestas-linguist/bin/linguist --breakdown .
100.00% HTML

HTML:
index.html
vendor/index.html

$
```

Now change the first line to `vendor/* linguist-detectable` and test again we get:

```
$ ../seppestas-linguist/bin/linguist --breakdown .
100.00% HTML

HTML:
index.html
vendor/docs/index.html
vendor/index.html

$
```

As you can see, the `linguist-documentation` attribute is being completely ignored.

Reply:
Mmh, true, though it kind of does that by design...

Would it be a good idea to fix this in `LazyBlob .detectable?`? I.e do something like
```
def detectable?
  if documentation? || generated? || vendored?
    return false
  elseif attr = git_attributes['linguist-detectable']
    return boolean_attribute(attr)
  else
    nil
  end
end
```

Reply:
Possibly, though I feel we're repeating ourselves unnecessarily.